### PR TITLE
Flash Resetting fixed it will maintain last set state now.

### PR DESCRIPTION
**Pull request by AliAyub007**
Fixes issue #292 

Changes:

Maintained last set flash state.

Screenshots for the change:

### CODE REVIEW

**Comment by phimpme-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">:tada: Thanks for the PR @AliAyub007 we hope you have changed the variables in the upload_apk_forks.sh its for easy moderation.</td>
    </tr>
  </tbody>
</table>

:link: Maintainers: the apk to test can be found at https://github.com/AliAyub007/phimpme-android/blob/apk/app-debug.apk

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 1]
```diff
@@ -944,7 +949,17 @@ public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
 		// the preview.
 		Log.e("Surface","Change");
 		Camera.Parameters parameters = mCamera.getParameters();
-        	parameters.setFlashMode(Camera.Parameters.FLASH_MODE_ON);
+		int state = Camera2.getState();
+		if(state == 1 ){
+			parameters.setFlashMode(Camera.Parameters.FLASH_MODE_OFF);
+		}
+		else if(state == 2)
+		{
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Please put the { (bracket) in the same line.

**Comment by vinaysajjanapu** [comment]
Already solved here #283

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 2]
```diff
@@ -115,6 +115,11 @@ public Camera2() {
 		super();
 
 	}
+	public static int getState()
+	{
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Can you pls make this inline. That is the bracket on the same line.

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 3]
```diff
@@ -944,7 +949,13 @@ public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
 		// the preview.
 		Log.e("Surface","Change");
 		Camera.Parameters parameters = mCamera.getParameters();
-        	parameters.setFlashMode(Camera.Parameters.FLASH_MODE_ON);
+		int state = Camera2.getState();
+		if(state == 1 )
+		{ parameters.setFlashMode(Camera.Parameters.FLASH_MODE_OFF); }
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Add this bracket also in the same line.

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 4]
```diff
@@ -944,7 +949,13 @@ public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
...
+		else if(state == 2)
+		{ parameters.setFlashMode(Camera.Parameters.FLASH_MODE_AUTO); }
+		else
+		{ parameters.setFlashMode(Camera.Parameters.FLASH_MODE_ON);	}
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Two more brackets.

File: app/src/main/java/vn/mbm/phimp/me/Camera2.java [hunk 5]
```diff
@@ -950,12 +950,9 @@ public void surfaceChanged(SurfaceHolder holder, int format, int w, int h) {
 		Log.e("Surface","Change");
 		Camera.Parameters parameters = mCamera.getParameters();
 		int state = Camera2.getState();
-		if(state == 1 )
-		{ parameters.setFlashMode(Camera.Parameters.FLASH_MODE_OFF); }
-		else if(state == 2)
-		{ parameters.setFlashMode(Camera.Parameters.FLASH_MODE_AUTO); }
-		else
-		{ parameters.setFlashMode(Camera.Parameters.FLASH_MODE_ON);	}
+		if(state == 1 )	{ parameters.setFlashMode(Camera.Parameters.FLASH_MODE_OFF); }
```
Comment (Abhi2424shek) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Move the if stmts body to another line. If you want you may not use brackets as well.

Comment (AliAyub007) [inline app/src/main/java/vn/mbm/phimp/me/Camera2.java]: Do let me know if there is any other problem. :)

**Comment by AliAyub007** [comment]
@Abhi2424shek  As @vinaysajjanapu  has solved it and it's merged so I think this PR should be closed.

**Pull request closed without merging**